mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -165,6 +166,38 @@ def __init__(self, name=None, bus=None, use_settings=True):
         # Skill Public API
         self.public_api = {}
 
+        self.skill_control = SkillControl()
+
+    def change_state(self, new_state):
+        self.log.debug("change_state() skill:%s - changing state from %s to %s" % (self.skill_id, self.skill_control.state, new_state))
+
+        if self.skill_control.states is None:
+            return
+
+        if new_state not in self.skill_control.states:
+            self.log.warning("invalid state change, from %s to %s" % (self.skill_control.state, new_state))
```

Comment:
this is a bit misleading.  the state change itself is not valid, the value of the new state is.  the log message doesn't say which the invalid value is, the old or the new.

Reply:
No, this is warning you the new state is NOT a valid value as it does not exist in your enumeration of states object (self.skill_control.states).